### DISCUSSION TITLE
How to use router query in boot file?

### DISCUSSION THREAD

**Discussion by mecjos** [quasarframework/quasar#9339] [General - Components / Directives / etc] [answered]
Hi, I want to check router query in a boot file as follows; 

```
window.Echo = new Echo({
  broadcaster: 'pusher',
  key: process.env.PUSHER_APP_KEY,
  authEndpoint: process.env.API + '/broadcasting/auth',
  auth: {headers: {Authorization: "Bearer " + getToken(route.query.token)}},
  cluster: process.env.PUSHER_CLUSTER,
  forceTLS: true
})
```

How can I reach route's query in a boot file 'out of vue component' ?? 

Thank you

**Comment by metalsadman** [comment, marked as answer]
@mecjos you have access to `router` in the default function as parameter https://quasar.dev/quasar-cli/boot-files#anatomy-of-a-boot-file.

**Reply by mecjos** [reply]
I couldn't understand how Can ı get current route's query from that doc page. Could you give an example?

**Reply by metalsadman** [reply]
```js
export default ({ app, router, store, Vue }) => {
  // something to do
  const query = router.currentRoute.query
}
```

**Reply by mecjos** [reply]
I tried that but currentRoute is empty in boot file. I think it's due to loading of before routes creation.

**Reply by stefanvanherwijnen** [reply]
I believe the boot files are run before the app is instantiated, thus there is no 'current route' in the boot file.
https://next.quasar.dev/quasar-cli/boot-files#quasar-app-flow
You could try setting a [navigation guard](https://next.router.vuejs.org/api/#navigationguard) in your boot file which will run when you access a route.
